sonar-core/src/main/java/org/sonar/core/util/logs/DefaultProfiler.java

```diff
@@ -108,70 +105,143 @@ public long stopInfo() {
   public long stopDebug(String message) {
-    return doStop(LoggerLevel.DEBUG, message, "");
+    return doStop(LoggerLevel.DEBUG, message, null, "");
+  }
+
+  @Override
+  public long stopDebug(String message, Object... args) {
+    return doStop(LoggerLevel.DEBUG, message, args, "");
   }
 
   @Override
   public long stopInfo(String message) {
-    return doStop(LoggerLevel.INFO, message, "");
+    return doStop(LoggerLevel.INFO, message, null, "");
   }
 
-  private long doStop(LoggerLevel level, @Nullable String message, String messageSuffix) {
+  @Override
+  public long stopInfo(String message, Object... args) {
+    return doStop(LoggerLevel.INFO, message, args, "");
+  }
+
+  private Profiler doStart(LoggerLevel logLevel, String message, Object... args) {
+    init(message, args);
+    logStartMessage(logLevel, message, args);
+    return this;
+  }
+
+  private void init(String message, Object... args) {
+    this.startTime = System2.INSTANCE.now();
+    this.startMessage = message;
+    this.args = args;
+  }
+
+  private void reset() {
+    this.startTime = 0L;
+    this.startMessage = null;
+    this.args = null;
+    this.context.clear();
+  }
+
+  private void logStartMessage(LoggerLevel loggerLevel, String message, Object... args) {
+    if (shouldLog(logger, loggerLevel)) {
+      StringBuilder sb = new StringBuilder();
+      sb.append(message);
+      appendContext(sb);
+      logger.trace(sb.toString(), args);
+    }
+  }
+
+  private long doStopWithoutMessage(LoggerLevel level) {
+    if (startMessage == null) {
+      throw new IllegalStateException("Profiler#stopXXX() can't be called without any message defined in start methods");
+    }
+    return doStop(level, startMessage, this.args, " (done)");
+  }
+
+  private long doStop(LoggerLevel level, @Nullable String message, @Nullable Object[] args, String messageSuffix) {
     if (startTime == 0L) {
       throw new IllegalStateException("Profiler must be started before being stopped");
     }
     long duration = System2.INSTANCE.now() - startTime;
-    StringBuilder sb = new StringBuilder();
-    if (!StringUtils.isEmpty(message)) {
-      sb.append(message);
-      sb.append(messageSuffix);
-      sb.append(CONTEXT_SEPARATOR);
+    if (shouldLog(logger, level)) {
+      StringBuilder sb = new StringBuilder();
+      if (!StringUtils.isEmpty(message)) {
+        sb.append(message);
+        sb.append(messageSuffix);
+        sb.append(CONTEXT_SEPARATOR);
+      }
+      sb.append("time=").append(duration).append("ms");
+      appendContext(sb);
+      log(level, sb.toString(), args);
     }
-    sb.append("time=").append(duration).append("ms");
-    appendContext(sb);
-    log(level, sb.toString());
-    startTime = 0L;
-    startMessage = null;
-    context.clear();
+    reset();
     return duration;
   }
 
-  void log(LoggerLevel level, String msg) {
+  void log(LoggerLevel level, String msg, @Nullable Object[] args) {
     switch (level) {
       case TRACE:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1151)